src/orca/services/nextflowtower/ops.py

```diff
@@ -104,17 +105,40 @@ def launch_workflow(
         self,
         launch_info: LaunchInfo,
         compute_env_filter: Optional[str] = None,
+        ignore_previous_runs: bool = False,
     ) -> str:
         """Launch a workflow using the latest matching compute env.
 
         Args:
             launch_info: Workflow launch information.
             compute_env_filter: Filter for matching compute
                 environments. Default to None.
+            ignore_previous_runs: Whether to ignore previous
+                workflow runs with the same attributes. Note
+                that enabling this might result in duplicate
+                workflow runs.
 
         Returns:
             Workflow run ID.
         """
+        # Make sure that essential attributes are set
+        if launch_info.pipeline is None or launch_info.run_name is None:
+            message = "LaunchInfo 'run_name' and 'pipeline' attributes must be set."
+            raise ValueError(message)
+
+        # Update launch_info if there are previous workflow runs
+        if not ignore_previous_runs:
```

Comment:
I would specify different run names. For example, you can name the runs based on the date (and time if run more than once per day). You could also set `ignore_previous_runs=True` to be more efficient (_i.e._ to avoid looking for previous workflow runs when you know there aren't any). However, that would prevent Orca from being able to re-run a failed workflow run. If the skipped files due to the failure would just get picked up the next day, then this isn't a concern, and you can set `ignore_previous_runs=True`.